taskcluster/scriptworker_taskgraph/transforms/cached.py

```diff
@@ -0,0 +1,63 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Build the cached_task digest to prevent rerunning tasks if the code hasn't changed.
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import hashlib
+import json
+import os
+import subprocess
+
+import taskgraph
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.hash import hash_paths
+from taskgraph.util.memoize import memoize
+
+transforms = TransformSequence()
+
+BASE_DIR = os.getcwd()
+
+@transforms.add
+def add_resources(config, tasks):
+    for task in tasks:
+        resources = task.pop("resources", None)
+        if resources:
+            task.setdefault("attributes", {})["resources"] = resources
```

Comment:
Nit: should we bail out if `task["attributes"]["resources"]` has already been defined before this point? If not, should we avoid overwriting it?

Reply:
Hm. We may inherit the k8s-image resources attribute from the tox kind, and intentionally overwrite.